recommender/engine_test.go

```diff
@@ -451,12 +438,21 @@ func TestEngine_RecommendCluster(t *testing.T) {
 			region:   "us-west-2",
 			check: func(response *ClusterRecommendationResp, err error) {
 				assert.Nil(t, err, "should not get error when recommending")
-				assert.NotNil(t, response, "the response shouldn't be nil")
+				assert.Equal(t, &ClusterRecommendationResp{Provider: "dummy", Zones: []string{"testZone1", "testZone2"},
+					NodePools: []NodePool{
```

Comment:
these deep equal statements are very hard to maintain; rather use direct assertions on the values